service/src/main/java/uk/gov/hmcts/reform/fpl/service/TaskListRenderer.java

```diff
@@ -0,0 +1,172 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.enums.Event;
+import uk.gov.hmcts.reform.fpl.model.tasklist.Task;
+import uk.gov.hmcts.reform.fpl.model.tasklist.TaskSection;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static java.util.List.of;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.enums.Event.ALLOCATION_PROPOSAL;
+import static uk.gov.hmcts.reform.fpl.enums.Event.APPLICANT;
+import static uk.gov.hmcts.reform.fpl.enums.Event.ATTENDING_THE_HEARING;
+import static uk.gov.hmcts.reform.fpl.enums.Event.CASE_NAME;
+import static uk.gov.hmcts.reform.fpl.enums.Event.CHILDREN;
+import static uk.gov.hmcts.reform.fpl.enums.Event.DOCUMENTS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.FACTORS_AFFECTING_PARENTING;
+import static uk.gov.hmcts.reform.fpl.enums.Event.GROUNDS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.HEARING_NEEDED;
+import static uk.gov.hmcts.reform.fpl.enums.Event.INTERNATIONAL_ELEMENT;
+import static uk.gov.hmcts.reform.fpl.enums.Event.ORDERS_NEEDED;
+import static uk.gov.hmcts.reform.fpl.enums.Event.OTHERS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.OTHER_PROCEEDINGS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.RESPONDENTS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.RISK_AND_HARM;
+import static uk.gov.hmcts.reform.fpl.enums.Event.SUBMIT_APPLICATION;
+import static uk.gov.hmcts.reform.fpl.model.tasklist.TaskSection.newSection;
+import static uk.gov.hmcts.reform.fpl.model.tasklist.TaskState.COMPLETED;
+import static uk.gov.hmcts.reform.fpl.model.tasklist.TaskState.NOT_AVAILABLE;
+
+@Service
+public class TaskListRenderer {
+
+    private final String imagesBaseUrl;
+
+    public TaskListRenderer(@Value("${resources.images.baseUrl}") String imagesBaseUrl) {
+        this.imagesBaseUrl = imagesBaseUrl;
+    }
+
+    //TODO consider templating solution like mustache
+    public String render(List<Task> allTasks) {
+        final List<String> lines = new LinkedList<>();
```

Comment:
Maybe move the declaration to be a bit closer to its usage, by the time I got to the bottom of the method I had completely forgotten about this variable and was slightly confused

Reply:
it means that method is too long, refactored to have separate method that returns tasks grouped by section